config/config.exs

```diff
@@ -28,4 +28,8 @@ config :phoenix, :json_library, Jason
 
 # Import environment specific config. This must remain at the bottom
 # of this file so it overrides the configuration defined above.
-import_config "#{config_env()}.exs"
+env = config_env()
+
+unless env == :docs do
```

Comment:
Actually, to remove the need for stuff like this, it is fine to keep ex_doc as a dev dependency too. :)